src/main/java/hlf/java/rest/client/listener/TransactionConsumer.java

```diff
@@ -89,14 +95,32 @@ public void listen(ConsumerRecord<String, String> message) {
           case FabricClientConstants.FABRIC_COLLECTION_NAME:
             collections = new String(msgHeader.value(), StandardCharsets.UTF_8);
             break;
+          case PAYLOAD_KIND:
+            payloadKind = new String(msgHeader.value(), StandardCharsets.UTF_8);
+            break;
           default:
             break;
         }
       }
 
-      if (!networkName.isEmpty()
-          && !contractName.isEmpty()
-          && !transactionFunctionName.isEmpty()
+      if (isIdentifiableFunction(networkName, contractName, transactionFunctionName)
+          && payloadKind.equals(PL_KIND_MULTI_DATA)) {
+
+        MultiDataTransactionPayload multiDataTransactionPayload;
+
+        try {
+          multiDataTransactionPayload =
+              objectMapper.readValue(transactionParams, MultiDataTransactionPayload.class);
+        } catch (Exception e) {
+          throw new ServiceException(
+              ErrorCode.CHANNEL_PAYLOAD_ERROR, "Invalid transaction payload provided");
+        }
+
+        transactionFulfillment.writeMultiDataTransactionToLedger(
+            networkName, contractName, transactionFunctionName, multiDataTransactionPayload);
+      }
+
+      if (isIdentifiableFunction(networkName, contractName, transactionFunctionName)
```

Comment:
There's a risk of both if conditions executed. Can you please either add more conditions here or break/return from the previous if condition?

Reply:
Right.. Adding return statement to previous invocation.